controllers/cache/generate.go

```diff
@@ -0,0 +1,82 @@
+package cache
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	redisCli "src/github.com/goharbor/harbor-cluster-operator/controllers/cache/client/api/v1"
+	"src/github.com/goharbor/harbor-cluster-operator/controllers/utils"
+)
+
+var (
+	virtualServiceGVR = schema.GroupVersionResource{
+		Group:    "databases.spotahome.com",
+		Version:  "v1",
+		Resource: "redisfailovers",
+	}
+)
+
+func (d *defaultCache) generateRedisCR() (*unstructured.Unstructured, error) {
+	name := d.Request.Name
+	nameSpace := d.Request.Namespace
+
+	resource := utils.GetRequests("1", "1Gi")
```

Comment:
After the harbor cluster operator spec is updated, this place needs to be modified.

Reply:
It should be created based on the Redis section of cluster spec.